src/main/java/org/jenkinsci/plugins/docker/traceability/model/DockerTraceabilityReportListener.java

```diff
@@ -76,21 +80,24 @@ public static void fire(@Nonnull DockerTraceabilityReport event) {
      * @param containerId Container ID (full 64-char representation)
      */
     public static void fireNewDeployment(@Nonnull String containerId) {
-        for (DockerEventListener listener : all()) {
+        for (DockerTraceabilityReportListener listener : all()) {
             try {
                 listener.onNewDeployment(containerId);
             } catch (Throwable t) { // Prevent failures on runtime exceptions
-                //TODO: logging
+                LOGGER.log(Level.SEVERE, "Runtime exception during the new deployment processing in "+ listener, t);
             }
         }
     }
 
     /**
      * Retrieves a list of Docker event listeners.
-     * @return A list of all {@link DockerEventListener} extensions.
+     * @return A list of all {@link DockerTraceabilityReportListener} extensions.
      */
-    public static @Nonnull ExtensionList<DockerEventListener> all() {
-        // TODO: null checks
-        return Jenkins.getInstance().getExtensionList(DockerEventListener.class);
+    public static @Nonnull ExtensionList<DockerTraceabilityReportListener> all() {
+        final Jenkins j = Jenkins.getInstance();
+        if (j == null) {
```

Comment:
why?
if jenkins is null we haven;t initialized and presumably bad things could happen in the implementation if things assume Jenkins is running?